src/memefactory/server/twitter_bot.cljs

```diff
@@ -25,24 +25,27 @@
    [goog.string :as gstring]
    [print.foo :refer [look] :include-macros true]
    [bignumber.core :as bn]
-   [memefactory.server.conversion-rates :as conv-rates]))
+   [memefactory.server.conversion-rates :as conv-rates]
+   [cljs.core.async :as async])
+  (:require-macros [memefactory.shared.utils :refer [safe-go <?]]))
 
 (def Twitter (nodejs/require "twitter"))
 (def fs (nodejs/require "fs"))
 (def tar-fs (nodejs/require "tar-fs"))
 
 (defn upload-file-to-twitter [twitter-obj file-content]
-  (js/Promise. (fn [resolve reject]
-                 (try
-                   (.post twitter-obj
-                          "media/upload"
-                          #js {:media file-content}
-                          (fn [error media response]
-                            (if error
-                              (reject error)
-                              (resolve (.-media_id_string media)))))
-                   (catch :default e
-                     (reject e))))))
+  (let [out-ch (async/promise-chan)]
+    (try
```

Comment:
why not use try-catch from https://github.com/district0x/district0x-error-handling (or put the error on the channel)?

Reply:
Good catch, returning the error on the chan now